pull out CSS color validator and validate <meta> theme-color

brendankenny opened the pull request:
also puts all of the current code to get `chrome-devtools-frontend` able to be imported into lighthouse (the other place was `network-recorder.js`) into one file for easier maintenance.

fixes #92

samccone commented on gatherers/theme-color.js:
```diff
@@ -26,7 +27,10 @@ class ThemeColor extends Gather {
       .then(node => node && node.getAttribute('content'))
       .then(themeColorMeta => {
-        return {themeColorMeta};
+        if (themeColorMeta && validColor(themeColorMeta)) {
```
we should test this behavior :) (I know it is not complex, but think of the test as ensuring the contract is held up)

brendankenny replied:
so setting up a test for this made it clear it was a bit weird to have in the gather phase. I've pushed it into the audit itself, and added tests for it there :)

brendankenny commented:
moved the valid meta theme-color check into the audit, added `debugString`s for missing meta theme-color tag, missing color on the tag, and invalid color in the tag, and added testing for an invalid color and the existence of warnings when the audit fails as above.

Since the audits are type checked, also added externs for `WebInspector.Color.parse`.

PTAL

paulirish commented:
v nice. wfm
